### PR TITLE
Migrated group session and group topic api's

### PR DESCRIPTION
**Pull request by UwaoEvan**
**What i did**
- Migrated group session and group topic api's

### CODE REVIEW

File: src/database/schema.ts
Comment (tindNan) [inline src/database/schema.ts:1]: I think you should revert this change

**Comment by tindNan** [comment]
`yarn format`. Please

I will release a github action to check for lint issues and run tests.

File: test/routes/groupSessions/fetch-group-sessions.test.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const sampleUser = await generateUser(app.db);
+  const token = await encodeAuthToken(sampleUser.id, "user");
+
+  const response = await app
+    .inject()
+    .headers({ authorization: `bearer ${token}` })
+    .get("/groupSessions");
+
+  t.equal(response.statusCode, 200);
+  t.ok(response.payload);
```
Comment (tindNan) [inline test/routes/groupSessions/fetch-group-sessions.test.ts:17]: I would review this test.

At this point the database does not have any group sessions, so how do we that we have returned the right thing?

Comment (UwaoEvan) [reply test/routes/groupSessions/fetch-group-sessions.test.ts:17]: The database will have the group sessions because the first test that runs and creates the group sessions in the db. This is the second test that runs after those sessions have been created.

Comment (tindNan) [reply test/routes/groupSessions/fetch-group-sessions.test.ts:17]: tests should be isolated unless they are described as part of a suite with subtests. A new developer reading these tests will have to navigate to another file to see where group sessions were added to the database

Comment (UwaoEvan) [reply test/routes/groupSessions/fetch-group-sessions.test.ts:21]: Updated as required.

Comment (tindNan) [reply test/routes/groupSessions/fetch-group-sessions.test.ts:21]: please also check the contents of the payload response match the group sessions that are in the `db`. You don't have to match all the keys, but checking the ids, topic ids etc match would be good.

File: test/routes/groupSessions/create-a-group-session.test.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    .inject()
+    .headers({ authorization: `Bearer ${token}` })
+    .post("/groupSessions")
+    .payload(payload);
+
+  const postedGroupSession = await app.db.query.groupSession.findFirst({
+    where: and(
+      eq(groupSession.groupTopicId, payload.groupTopicId),
+      eq(groupSession.therapistId, payload.therapistId),
+      eq(sql`DATE(${groupSession.startTime})`, payload.startTime),
```
Comment (tindNan) [inline test/routes/groupSessions/create-a-group-session.test.ts:42]: I think here we want to match on the date + time since that's important for the session

Comment (UwaoEvan) [reply test/routes/groupSessions/create-a-group-session.test.ts:42]: No only the date because on the same date we can have more than one sessions.

Comment (tindNan) [reply test/routes/groupSessions/create-a-group-session.test.ts:42]: ah okay.

Well `payload.startTime` has a time component to it so I am a bit confused there.

Comment (UwaoEvan) [reply test/routes/groupSessions/create-a-group-session.test.ts:42]: Noted

**Review by tindNan — APPROVED** [review]
